src/error.rs

```diff
@@ -67,35 +71,61 @@ impl fmt::Debug for ErrorCode {
     }
 }
 
-#[derive(Debug, PartialEq, Eq, Clone)]
+#[derive(Debug)]
 pub enum Error {
-    JSON(String),
-    Utf8(String),
-    IOError(String),
-    InvalidOption(String),
-    HWIError(String, Option<ErrorCode>),
-    PyErr(String),
+    Json(serde_json::Error),
+    Utf8(std::str::Utf8Error),
+    Io(std::io::Error),
+    Hwi(String, Option<ErrorCode>),
+    Python(pyo3::PyErr),
 }
 
-macro_rules! impl_error {
-    ( $from:ty, $to:ident ) => {
-        impl std::convert::From<$from> for Error {
-            fn from(err: $from) -> Self {
-                Error::$to(err.to_string())
-            }
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use Error::*;
+
+        match *self {
+            Json(_) => f.write_str("serde_json error"),
+            Utf8(_) => f.write_str("utf8 error"),
+            Io(_) => f.write_str("I/O error"),
+            Hwi(ref s, ref code) => write!(f, "HWI error: {}, ({:?})", s, code),
+            Python(_) => f.write_str("python error"),
         }
-    };
+    }
 }
 
-impl_error!(serde_json::Error, JSON);
-impl_error!(std::str::Utf8Error, Utf8);
-impl_error!(std::io::Error, IOError);
-impl_error!(pyo3::prelude::PyErr, PyErr);
+#[cfg(feature = "std")]
+impl std::error::Error for Error {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        use self::Error::*;
 
-impl fmt::Display for Error {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "{:?}", self)
+        match self {
+            UtxoMissing(_) => None,
```

Comment:
I'm a bit confused, there's no `UtxoMissing` variant in `Error`. I think this piece of code is not being compiled because of the `#[cfg(feature = "std")]` just above

Reply:
Looks like I cut'n paste the whole function and forgot to change it. My bad.